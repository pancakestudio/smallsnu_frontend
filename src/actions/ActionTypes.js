export const BUILDING_CLICK = "BUILDING_CLICK";
