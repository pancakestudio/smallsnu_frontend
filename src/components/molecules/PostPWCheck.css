.PostPwCheck {
  min-width: 30vw;
}
