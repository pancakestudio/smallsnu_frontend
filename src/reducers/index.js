import { combineReducers } from 'redux';
import buildingButton from './buildingButton';

const reducers = combineReducers({
  buildingButton
});

export default reducers;
