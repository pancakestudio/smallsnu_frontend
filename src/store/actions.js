import * as types from './actionTypes'

/******** Map ********/

export function requestBuilding(bldgNo){
  return{
    type: types.REQUEST_BUILDING,
    bldgNo
  };
}

export function getBuildingSuccess(krName, bldgNo, info, restaurants, seminars, lectures, posts){
  return{
    type: types.GET_BUILDING_SUCCESS,
    krName,
    bldgNo,
    info,
    restaurants,
    seminars,
    lectures,
    posts,
  }
}

export function getBuildingFailure(error){
  return{
    type: types.GET_BUILDING_FAILURE,
    error
  }
}

export function zoomChanged(zoomLevel){
  return {
    type: types.ZOOM_CHANGED,
    zoomLevel
  }
}

export function hideMarkers(){
  return {
    type: types.HIDE_MARKERS
  }
}

/******** SideBar ********/

export function toggleSideBar(){
  return {
    type: types.TOGGLE_SIDE_BAR
  }
}

/******** SearchBar ********/

export function searchValueChange(bldgNo){
  return{
    type: types.SEARCH_VALUE_CHANGE,
    bldgNo
  };
}

export function search(bldgNo, bldgPos){
  return{
    type: types.SEARCH,
    bldgNo,
    bldgPos
  }
}

/******** Post ********/

export function requestBoard(bldgNo){
  return {
    type: types.REQUEST_BOARD,
    bldgNo
  }
}

export function getBoardSuccess(posts, bldgNo){
  return {
    type: types.GET_BOARD_SUCCESS,
    posts,
    bldgNo
  }
}

export function getBoardFailure(error){
  return {
    type: types.GET_BOARD_FAILURE,
    error
  }
}

export function requestPost(id) {
  return {
    type: types.REQUEST_POST,
    id
  }
}

export function getPostSuccess(post){
  return {
    type: types.GET_POST_SUCCESS,
    post
  }
}

export function getPostFailure(error){
  return {
    type: types.GET_POST_FAILURE,
    error
  }
}

export function deletePost(post, bldgNo){
  return{
    type: types.DELETE_POST,
    post,
    bldgNo,
  }
}

export function savePost(post, bldgNo){
  return{
    type: types.SAVE_POST,
    post,
    bldgNo
  }
}

export function editPost(post, bldgNo){
  return{
    type: types.EDIT_POST,
    post,
    bldgNo
  }
}

export function editPostFlag(){
  return{
    type: types.EDIT_POST_FLAG,
  }
}

export function showWritePost() {
  return{
    type: types.SHOW_WRITE_POST,
  };
}

export function showPostPWCheck(){
  return{
    type: types.SHOW_POST_PW_CHECK
  }
}

export function hidePostPWCheck(){
  return{
    type: types.HIDE_POST_PW_CHECK
  }
}

export function postPWCheck(password) {
  return{
    type: types.POST_PW_CHECK,
    password
  }
}

export function changeBoardPage(page){
  return {
    type: types.CHANGE_BOARD_PAGE,
    page
  }
}

/******** Restaurant ********/

export function toggleResMarker(){
  return{
    type: types.TOGGLE_RES_MARKER,
  }
}

export function requestRestaurant(id){
  return{
    type: types.REQUEST_RESTAURANT,
    id
  }
}

export function getRestaurantSuccess(restaurant){
  return{
    type : types.GET_RESTAURANT_SUCCESS,
    restaurant,
  }
}

export function getRestaurantFailure(error){
  return{
    type: types.GET_RESTAURANT_FAILURE,
    error
  }
}

export function requestAllRestaurants(){
  return{
    type: types.REQUEST_ALL_RESTAURANTS,
  }
}


export function getAllRestaurantsSuccess(restaurants){
  return{
    type : types.GET_ALL_RESTAURANTS_SUCCESS,
    restaurants,
  }
}

export function getAllRestaurantsFailure(error){
  return{
    type: types.GET_ALL_RESTAURANTS_FAILURE,
    error
  }
}

/******** Seminar ********/

export function toggleSemiMarker(){
  return{
    type: types.TOGGLE_SEMI_MARKER,
  }
}

export function changeSeminarPage(page){
  return {
    type: types.CHANGE_SEMINAR_PAGE,
    page
  }
}

export function requestSeminar(id){
  return {
    type: types.REQUEST_SEMINAR,
    id
  }
}

export function getSeminarSuccess(seminar){
  return {
    type: types.GET_SEMINAR_SUCCESS,
    seminar
  }
}

export function getSeminarFailure(error){
  return {
    type: types.GET_SEMINAR_FAILURE,
    error
  }
}

export function requestBldgSeminars(bldgNo){
  return {
    type: types.REQUEST_BLDG_SEMINARS,
    bldgNo
  }
}

export function getBldgSeminarsSuccess(seminars, bldgNo){
  return {
    type: types.GET_BLDG_SEMINARS_SUCCESS,
    seminars,
    bldgNo
  }
}

export function getBldgSeminarsFailure(error){
  return {
    type: types.GET_BLDG_SEMINARS_FAILURE,
    error
  }
}

export function requestAllSeminars(){
  return {
    type: types.REQUEST_ALL_SEMINARS,
  }
}

export function getAllSeminarsSuccess(seminars){
  return {
    type: types.GET_ALL_SEMINARS_SUCCESS,
    seminars
  }
}

export function getAllSeminarsFailure(error){
  return {
    type: types.GET_ALL_SEMINARS_FAILURE,
    error
  }
}

/******** Bank ********/

export function toggleBankMarker(){
  return{
    type: types.TOGGLE_BANK_MARKER,
  }
}

export function requestAllBanks(){
  return{
    type: types.REQUEST_ALL_BANKS,
  }
}

export function getAllBanksSuccess(banks){
  return{
    type: types.GET_ALL_BANKS_SUCCESS,
    banks,
  }
}

export function getAllBanksFailure(error){
  return{
    type: types.GET_ALL_BANKS_FAILURE,
    error,
  }
}

export function requestBank(id){
  return{
    type: types.REQUEST_BANK,
    id
  }
}

export function getBankSuccess(bank){
  return{
    type: types.GET_BANK_SUCCESS,
    bank
  }
}

export function getBankFailure(error){
  return{
    type: types.GET_BANK_FAILURE,
    error
  }
}


/******** ATM ********/

export function toggleATMMarker(){
  return{
    type: types.TOGGLE_ATM_MARKER,
  }
}

export function requestAllATMs(){
  return{
    type: types.REQUEST_ALL_ATMS,
  }
}

export function getAllATMsSuccess(atms){
  return{
    type: types.GET_ALL_ATMS_SUCCESS,
    atms,
  }
}

export function getAllATMsFailure(error){
  return{
    type: types.GET_ALL_ATMS_FAILURE,
    error,
  }
}

export function requestATM(id){
  return{
    type: types.REQUEST_ATM,
    id
  }
}

export function getATMSuccess(atm){
  return{
    type: types.GET_ATM_SUCCESS,
    atm
  }
}

export function getATMFailure(error){
  return{
    type: types.GET_ATM_FAILURE,
    error
  }
}

/******** Cafe ********/

export function toggleCafeMarker(){
  return{
    type: types.TOGGLE_CAFE_MARKER,
  }
}

export function requestAllCafes(){
  return{
    type: types.REQUEST_ALL_CAFES,
  }
}

export function getAllCafesSuccess(cafes){
  return{
    type: types.GET_ALL_CAFES_SUCCESS,
    cafes,
  }
}

export function getAllCafesFailure(error){
  return{
    type: types.GET_ALL_CAFES_FAILURE,
    error,
  }
}

export function requestCafe(id){
  return{
    type: types.REQUEST_CAFE,
    id
  }
}

export function getCafeSuccess(cafe){
  return{
    type: types.GET_CAFE_SUCCESS,
    cafe
  }
}

export function getCafeFailure(error){
  return{
    type: types.GET_CAFE_FAILURE,
    error
  }
}


/******** Conv ********/

export function toggleConvMarker(){
  return{
    type: types.TOGGLE_CONV_MARKER,
  }
}

export function requestAllConves(){
  return{
    type: types.REQUEST_ALL_CONVES,
  }
}

export function getAllConvesSuccess(conves){
  return{
    type: types.GET_ALL_CONVES_SUCCESS,
    conves,
  }
}

export function getAllConvesFailure(error){
  return{
    type: types.GET_ALL_CONVES_FAILURE,
    error,
  }
}

export function requestConv(id){
  return{
    type: types.REQUEST_CONV,
    id
  }
}

export function getConvSuccess(conv){
  return{
    type: types.GET_CONV_SUCCESS,
    conv
  }
}

export function getConvFailure(error){
  return{
    type: types.GET_CONV_FAILURE,
    error
  }
}
