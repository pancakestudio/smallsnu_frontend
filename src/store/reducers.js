import { combineReducers } from 'redux';
import * as types from './actionTypes'

/******** Map ********/

function currentPos(state = {lat: 37.459, lng: 126.952}, action){
  switch(action.type){
    case types.SEARCH:
      return action.bldgPos
    default:
      return state
  }
}

function zoom(state = 17, action) {
  switch(action.type){
    case types.TOGGLE_RES_MARKER:
    case types.TOGGLE_SEMI_MARKER:
      return 15
    case types.ZOOM_CHANGED:
      return action.zoomLevel
    default:
      return state
  }
}

function selectedBldg(state = {}, action){
  switch(action.type){
    case types.GET_BUILDING_SUCCESS:
      return Object.assign({}, state, {
        krName: action.krName,
        bldgNo: action.bldgNo,
        info: action.info,
        restaurants: action.restaurants,
        seminars: action.seminars,
        posts: action.posts
      })
    case types.GET_BOARD_SUCCESS:
      return Object.assign({}, state, {
        posts: action.posts
      })
    default:
      return state
  }
}

function showSearchMarker(state = false, action){
  switch(action.type){
    case types.SEARCH:
      return true
    default:
      return false
  }
}

function showResMarkers(state = false, action){
  switch(action.type){
    case types.TOGGLE_RES_MARKER:
      return !state
    case types.GET_ALL_RESTAURANTS_FAILURE:
    case types.HIDE_MARKERS:
    case types.TOGGLE_SEMI_MARKER:
    case types.TOGGLE_BANK_MARKER:
    case types.TOGGLE_ATM_MARKER:
    case types.TOGGLE_CAFE_MARKER:
    case types.TOGGLE_CONV_MARKER:
      return false
    default:
      return state
  }
}

function showBankMarkers(state = false, action){
  switch(action.type){
    case types.TOGGLE_BANK_MARKER:
      return !state
    case types.GET_ALL_BANKS_FAILURE:
    case types.HIDE_MARKERS:
    case types.TOGGLE_SEMI_MARKER:
    case types.TOGGLE_RES_MARKER:
    case types.TOGGLE_ATM_MARKER:
    case types.TOGGLE_CAFE_MARKER:
    case types.TOGGLE_CONV_MARKER:
      return false
    default:
      return state
  }
}

function showATMMarkers(state = false, action){
  switch(action.type){
    case types.TOGGLE_ATM_MARKER:
      return !state
    case types.GET_ALL_ATMS_FAILURE:
    case types.HIDE_MARKERS:
    case types.TOGGLE_SEMI_MARKER:
    case types.TOGGLE_RES_MARKER:
    case types.TOGGLE_BANK_MARKER:
    case types.TOGGLE_CAFE_MARKER:
    case types.TOGGLE_CONV_MARKER:
      return false
    default:
      return state
  }
}

function showCafeMarkers(state = false, action){
  switch(action.type){
    case types.TOGGLE_CAFE_MARKER:
      return !state
    case types.GET_ALL_CAFES_FAILURE:
    case types.HIDE_MARKERS:
    case types.TOGGLE_SEMI_MARKER:
    case types.TOGGLE_RES_MARKER:
    case types.TOGGLE_BANK_MARKER:
    case types.TOGGLE_ATM_MARKER:
    case types.TOGGLE_CONV_MARKER:
      return false
    default:
      return state
  }
}

function showConvMarkers(state = false, action){
  switch(action.type){
    case types.TOGGLE_CONV_MARKER:
      return !state
    case types.GET_ALL_CONVES_FAILURE:
    case types.HIDE_MARKERS:
    case types.TOGGLE_SEMI_MARKER:
    case types.TOGGLE_RES_MARKER:
    case types.TOGGLE_BANK_MARKER:
    case types.TOGGLE_ATM_MARKER:
    case types.TOGGLE_CAFE_MARKER:
      return false
    default:
      return state
  }
}

function showSemiMarkers(state = false, action){
  switch(action.type){
    case types.TOGGLE_SEMI_MARKER:
      return !state
    case types.GET_ALL_SEMINARS_FAILURE:
    case types.HIDE_MARKERS:
    case types.TOGGLE_RES_MARKER:
      return false
    default:
      return state
  }
}

function searchedBldg(state = "0", action){
  switch(action.type){
    case types.SEARCH:
      return action.bldgNo
    default:
      return state
  }
}

/******** SideBar ********/

function showSideBar(state = false, action){
  switch(action.type){
    case types.TOGGLE_SIDE_BAR:
      return !state
    default:
      return state
  }
}

/******** SearchBar ********/

function searchingBldg(state = "0", action){
  switch(action.type){
    case types.SEARCH_VALUE_CHANGE:
      return action.bldgNo
    default:
      return state
  }
}


/******** Post ********/

function showWritePostModal(state = false, action){
  switch(action.type){
    case types.SHOW_WRITE_POST:
      return true
    default:
      return false
  }
}

function selectedBoardBldgNo(state = "0", action){
  switch(action.type){
    case types.GET_BOARD_SUCCESS:
    case types.GET_BUILDING_SUCCESS:
      return action.bldgNo
    default:
      return state
  }
}

function selectedPostList(state = [], action){
  switch(action.type){
    case types.GET_BOARD_SUCCESS:
      return action.posts
    default:
      return state
  }
}

function selectedPost(state = {}, action){
  switch(action.type){
    case types.GET_POST_SUCCESS:
      return action.post
    default:
      return state
  }
}

function isEdit(state = false, action){
  switch(action.type){
    case types.EDIT_POST_FLAG:
      return true
    case types.GET_BOARD_SUCCESS:
      return false
    default:
      return state
  }
}

function postPW(state = "", action){
  switch(action.type){
    case types.POST_PW_CHECK:
      return action.password
    case types.GET_BOARD_SUCCESS:
      return ""
    default:
      return state
  }
}

function showPostPWCheck(state = false, action){
  switch(action.type){
    case types.SHOW_POST_PW_CHECK:
      return true
    default:
      return false
  }
}

function activeBoardPage(state = 1, action) {
  switch(action.type){
    case types.CHANGE_BOARD_PAGE:
      return action.page
    case types.GET_BOARD_SUCCESS:
      return 1
    default:
      return state
  }
}
/******** Bank ********/

function allBanks(state = [], action){
  switch(action.type){
    case types.GET_ALL_BANKS_SUCCESS:
      return action.banks
    default:
      return state
  }
}

function selectedBank(state = {}, action){
  switch(action.type){
    case types.GET_BANK_SUCCESS:
      return action.bank
    default:
      return state
  }
}

/******** ATM ********/

function allATMs(state = [], action){
  switch(action.type){
    case types.GET_ALL_ATMS_SUCCESS:
      return action.atms
    default:
      return state
  }
}

function selectedATM(state = {}, action){
  switch(action.type){
    case types.GET_ATM_SUCCESS:
      return action.atm
    default:
      return state
  }
}

/******** Cafe ********/

function allCafes(state = [], action){
  switch(action.type){
    case types.GET_ALL_CAFES_SUCCESS:
      console.log(action.cafes)
      return action.cafes
    default:
      return state
  }
}

function selectedCafe(state = {}, action){
  switch(action.type){
    case types.GET_CAFE_SUCCESS:
      return action.cafe
    default:
      return state
  }
}

/******** Convenient Store ********/

function allConves(state = [], action){
  switch(action.type){
    case types.GET_ALL_CONVES_SUCCESS:
      return action.conves
    default:
      return state
  }
}

function selectedConv(state = {}, action){
  switch(action.type){
    case types.GET_CONV_SUCCESS:
      return action.conv
    default:
      return state
  }
}

/******** Restaurant ********/

function selectedRes(state = {}, action){
  switch(action.type){
    case types.GET_RESTAURANT_SUCCESS:
      return action.restaurant
    default:
      return state
  }
}

function allRestaurants(state = [], action){
  switch(action.type){
    case types.GET_ALL_RESTAURANTS_SUCCESS:
      return action.restaurants
    default:
      return state
  }
}

/******** Seminar ********/

function selectedSemi(state = {}, action){
  switch(action.type){
    case types.GET_SEMINAR_SUCCESS:
      return action.seminar
    default:
      return state
  }
}

function selectedSemiList(state = [], action){
  switch(action.type){
    case types.GET_BLDG_SEMINARS_SUCCESS:
      return action.seminars
    default:
      return state
  }
}

function selectedSemiListBldgNo(state = "0", action){
  switch(action.type){
    case types.GET_BLDG_SEMINARS_SUCCESS:
      return action.bldgNo
    default:
      return state
  }
}

function activeSemiPage(state = 1, action) {
  switch(action.type){
    case types.CHANGE_SEMINAR_PAGE:
      return action.page
    case types.GET_BLDG_SEMINARS_SUCCESS:
      return 1
    default:
      return state
  }
}


function allSeminars(state = [], action) {
  switch(action.type){
    case types.GET_ALL_SEMINARS_SUCCESS:
      return action.seminars
    default:
      return state
  }
}

/******** App ********/

function error(state = "", action){
  switch(action.type){
    case types.GET_BUILDING_FAILURE:
    case types.GET_BOARD_FAILURE:
    case types.GET_POST_FAILURE:
    case types.GET_RESTAURANT_FAILURE:
    case types.GET_ALL_RESTAURANTS_FAILURE:
    case types.GET_SEMINAR_FAILURE:
    case types.GET_BLDG_SEMINARS_FAILURE:
    case types.GET_ALL_SEMINARS_FAILURE:
      return action.error
    default:
      return ""
  }
}

const reducers = combineReducers({

  currentPos, zoom, selectedBldg, showSearchMarker, showResMarkers,
  showSemiMarkers, searchedBldg, showBankMarkers, showATMMarkers, showCafeMarkers,
  showConvMarkers,// Map

  showSideBar, // SideBar
  searchingBldg, // SearchBar

  showWritePostModal, selectedBoardBldgNo, selectedPostList, selectedPost,
  isEdit, postPW, showPostPWCheck, activeBoardPage, // Post

  allBanks,selectedBank, // Bank
  allATMs, selectedATM, // ATM
  allCafes, selectedCafe, // Cafe
  allConves, selectedConv,  // Convenient Store
  selectedRes, allRestaurants, // Restaurant

  selectedSemi, selectedSemiList, selectedSemiListBldgNo,
  activeSemiPage, allSeminars, // Seminar

  error // App

});

export default reducers
