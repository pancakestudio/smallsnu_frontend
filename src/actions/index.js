import * as types from './ActionTypes';

export function buildingClick(e){
  return{
    type: types.BUILDING_CLICK,
    e
  };
}
