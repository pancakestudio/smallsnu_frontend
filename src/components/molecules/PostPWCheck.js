import React from 'react';
import { Modal, Button, Form } from 'react-bootstrap'
import { FaAngleLeft } from 'react-icons/fa'
import { historyPush } from '../../utils/Functions'
import './PostPWCheck.css'

export const PostPWCheck = ({show, post, bldgNo, onHideModal, onDelete}) => {
  let passwordInput
  const handleSubmit = () => {
    if(post.password === passwordInput.value){
      onDelete(post, bldgNo)
      onHideModal()
    }else{
      alert('비밀번호가 틀렸습니다')
    }
  }
  const handleBack = () => {
    onHideModal()
  }

  return(
    <Modal
      show = {show}
      onHide = {() =>{historyPush('/')}}
      dialogClassName = "PostPWCheck"
      centered
    >
      <Modal.Header closeButton>
      <Button className="back" onClick={handleBack}><FaAngleLeft /></Button>
        <Modal.Title>비밀번호 확인</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <Form.Group controlId="formGridPassword">
          <Form.Control type="password" placeholder="비밀번호" ref = {(ref) => (passwordInput=ref)}/>
        </Form.Group>
      </Modal.Body>
      <Modal.Footer>
        <Button variant="primary" onClick = {handleSubmit}>확인</Button>
      </Modal.Footer>
    </Modal>
  )
}
