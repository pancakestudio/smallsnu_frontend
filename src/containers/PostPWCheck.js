import { connect } from 'react-redux'
import { PostPWCheck } from '../components/molecules/PostPWCheck'
import { hidePostPWCheck, deletePost } from '../store/actions'

const mapStateToProps = (state) => {
  return {
    show: state.showPostPWCheck,
    post: state.selectedPost,
    bldgNo: state.selectedBoardBldgNo,
  }
}

const mapDispatchToProps = (dispatch) => {
  return{
    onHideModal : () => {
      dispatch(hidePostPWCheck())
    },
    onDelete : (post, bldgNo) => {
      dispatch(deletePost(post, bldgNo))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(PostPWCheck)
