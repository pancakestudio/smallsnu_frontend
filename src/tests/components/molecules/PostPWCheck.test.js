import React from 'react'
import configureStore from 'redux-mock-store'
import { shallow, mount } from 'enzyme'
import { Provider } from 'react-redux'
import ConnectedPostPWCheck from '../../../containers/PostPWCheck'
import { createStore } from 'redux'
import { PostPWCheck } from '../../../components/molecules/PostPWCheck'
import * as actions from '../../../store/actions'
import reducers from '../../../store/reducers'

describe('PostPWCheck', () => {
  global.window = { location: { pathname: null } };
  let component
  const mockDelete = jest.fn()
  const mockHideModal = jest.fn()
  const post = {
    id: 1,
    title: 'title',
    content: 'content',
    username: 'user',
    password: '1234'
  }
  it('renders correctly', () => {
    component = shallow(
      <PostPWCheck
        show = {true}
        post = {post}
        bldgNo={'301'}
        onDelete = {mockDelete}
      />
    )
  })

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('has form, form control, and button', ()=>{
    expect(component.find('FormControl').exists()).toBe(true)
    expect(component.find('Button').exists()).toBe(true)
    expect(component.find('ModalTitle').text()).toBe('비밀번호 확인')
  })

  it('calls functions', () => {
    component.find('Bootstrap(Modal)').simulate('hide')
    expect(global.window.location.pathname).toEqual('/')
  })

  it('correct password', () => {

  })
})

describe('ConnectedPostPWCheck', () => {
  const mockStore = configureStore()
  let store, component
  const initialState = {
    showPostPWCheck: true,
    selectedPost: {
      id: 1,
      title: 'title',
      content: 'content',
      username: 'user',
      password: '1234'
    },
    selectedBoardBldgNo: '301'
  }
  const post = {
    id: 1,
    title: 'title',
    content: 'content',
    username: 'user',
    password: '1234'
  }

  it('renders correctly', () => {
    store = mockStore(initialState)
    component = mount(<Provider store = {store}><ConnectedPostPWCheck/></Provider>)
  })

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot()
  })

  it('has a modal', ()=>{
    expect(component.find('Bootstrap(Modal)').exists()).toBe(true)
  })

  it('correct password', () => {
    component.find('input').instance().value = '1234'
    component.find('Button').simulate('click')
    expect(store.getActions()[0]).toEqual(actions.deletePost(post,'301'))
    expect(store.getActions()[1]).toEqual(actions.hidePostPWCheck())
  })

  it('incorrect password', () => {
    window.alert = jest.fn()
    component.find('input').instance().value = '1'
    component.find('Button').simulate('click')
    expect(window.alert).toHaveBeenCalledWith('비밀번호가 틀렸습니다')
  })

})
