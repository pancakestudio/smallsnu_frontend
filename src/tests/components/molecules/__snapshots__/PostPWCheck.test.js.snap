// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedPostPWCheck matches snapshot 1`] = `ReactWrapper {}`;

exports[`PostPWCheck matches snapshot 1`] = `ShallowWrapper {}`;
